components/script/dom/webgpu/gpuadapter.rs

```diff
@@ -37,7 +38,7 @@ pub(crate) struct GPUAdapter {
     channel: WebGPU,
     name: DOMString,
     #[ignore_malloc_size_of = "mozjs"]
-    extensions: Heap<*mut JSObject>,
+    extensions: RootedTraceableBox<Heap<*mut JSObject>>,
```

Comment:
This introduces redundant rooting (heap is already rooted via GPUAdapter) on each use.